frontend/src/components/accountPage/UserListLogic/UserListLogic.tsx

```diff
@@ -0,0 +1,192 @@
+  });
+
+  useEffect(() => {
+    abortUploadingRef.current = abortUploading;
+  }, [abortUploading]);
+
+  const handleAbort = (value: boolean) => setAbortUploading(value);
+  useEffect(() => {
+    if (!loading) {
+      setUsers(
+        data?.tenants?.items?.map(user => {
+          return {
+            key: user?.metadata?.tenantId!,
+            userid: user?.metadata?.tenantId!,
+            name: user?.spec?.firstName!,
+            surname: user?.spec?.lastName!,
+            email: user?.spec?.email!,
+            currentRole: user?.spec?.workspaces?.filter(roles =>
+              workspaceName.includes(roles?.workspaceRef?.name!)
+            )[0]?.role!,
+            workspaces:
+              user?.spec?.workspaces?.map(workspace => ({
+                role: workspace?.role as string,
+                name: workspace?.workspaceRef?.name! as string,
+              })) || [],
+          };
+        }) || []
+      );
+    }
+  }, [loading, data, workspaceName]);
+
+  const [applyTenantMutation] = useApplyTenantMutation();
+
+  const updateUser = async (user: User, newRole: string) => {
+    try {
+      let workspaces = users
+        .filter(u => u.userid === user.userid)[0]
+        .workspaces.map(w =>
+          w.name === workspaceName.split('-')[1]
+            ? { ...w, role: newRole as any }
+            : w
+        );
+
+      await applyTenantMutation({
+        variables: {
+          tenantId: user.userid,
+          patchJson: getTenantPatchJson({}, workspaces),
+        },
+      });
+      setUsers(
+        users.map(u =>
+          u.userid === user.userid
+            ? {
+                ...u,
+                currentRole: newRole,
+                workspaces: workspaces,
+              }
+            : u
+        )
+      );
+    } catch (error) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const addUser = async (usersToAdd: User[], workspaces?: any) => {
+    try {
+      if (workspaces) {
+        await applyTenantMutation({
+          variables: {
+            tenantId: usersToAdd[0].userid,
+            patchJson: getTenantPatchJson({}, workspaces),
+          },
+        });
+
+        usersToAdd[0].workspaces = workspaces;
+        setUsers([...users, usersToAdd[0]]);
+      } else {
+        setUploadedNumber(0);
+        // add multiple users from CSV
+        // for (const user of usersToAdd) {
+        //   await applyTenantMutation({
+        //     variables: {
+        //       tenantId: user.userid,
+
+        //       patchJson: getTenantPatchJson(
+        //         {
+        //           email: user.email,
+        //           firstName: user.name,
+        //           lastName: user.surname,
+        //         },
+        //         workspaces
+        //       ),
+        //     },
+        //   });
+        // }
+        let usersAdded: User[] = [];
+        for (const user of usersToAdd) {
+          if (abortUploadingRef.current) {
+            break;
+          } else {
```

Comment:
```suggestion
          if (abortUploadingRef.current) break;
```
else is not required since you're breaking if the condition is true...